internal/backend/scheduler/scheduler.go

```diff
@@ -109,11 +113,29 @@ func (sch *Scheduler) Update(ctx context.Context, tid sdktypes.TriggerID, schedu
 func (sch *Scheduler) workflow(wctx workflow.Context, tid sdktypes.TriggerID) error {
 	sl := sch.sl.With("trigger_id", tid)
 
-	event := sdktypes.NewEvent(tid).WithType("tick")
-
 	ctx := temporalclient.NewWorkflowContextAsGOContext(wctx)
 
-	eid, err := sch.dispatcher.Dispatch(ctx, event, nil)
+	// It's ok to run all these straight in the workflow - they're all should be
+	// pretty fast.
+
+	t, err := sch.triggers.Get(ctx, tid)
+	if err != nil {
+		if !errors.Is(err, sdkerrors.ErrNotFound) {
+			return fmt.Errorf("get trigger: %w", err)
+		}
+	}
+
+	if !t.IsValid() {
+		sl.Warnf("trigger %v not found, removing schedule", tid)
+
+		if err := sch.Delete(ctx, tid); err != nil {
+			return fmt.Errorf("delete schedule: %w", err)
+		}
+
+		return nil
+	}
+
+	eid, err := sch.dispatcher.Dispatch(ctx, sdktypes.NewEvent(tid).WithType("tick"), nil)
```

Comment:
this is slow, how does it even work in the workflow itself ?

Reply:
it's not slow - it's starting a new workflow.